### PR TITLE
II-17: Initializing wCurKeys and wNewKeys

### LINKED ISSUE #75: II-17 Input: Missing initialization of `wCurKeys` and `wNewKeys`
Was actually kind of helpful to have this left as an exercise for the reader but it'd be nice to include a note.

### PR DESCRIPTION
**Pull request by kav**
Fixes #75

### CODE REVIEW

**Review by avivace — APPROVED** [review]

**Comment by eievui5** [comment]
These `ld`s should be on the other side of the "Main" anchor

**Comment by eievui5** [comment]
Oh, there isn't an "init" anchor for this chapter. I think the chapter itself needs to be corrected, not just this code

**Comment by kav** [comment]
Yeah I put them there so they'd be set as part of updating main but I can add an init section instead after we define them if you'd prefer

**Comment by eievui5** [comment]
Yeah, that sounds best. Something like this:

> We also need to initialize these when our game starts, so add two more lines:
> ```
> 	; Turn the LCD on
> 	ld a, LCDCF_ON | LCDCF_BGON | LCDCF_OBJON
> 	ld [rLCDC], a
> 
> 	; During the first (blank) frame, initialize display registers
> 	ld a, %11100100
> 	ld [rBGP], a
> 	ld a, %11100100
> 	ld [rOBP0], a
> 
> 	ld a, 0
> 	ld [wFrameCounter], a
> 	ld [wCurKeys], a
> 	ld [wNewKeys], a
> ```

I think diff syntax would help highlight the new lines but I'm not sure if the leading + signs would be confusing--do we use diffs anywhere else?

**Comment by kav** [comment]
Diff syntax doesn't show up until adding the brick check two sections later.
```
+	call CheckAndHandleBrick
```

I'm inclined to instead introduce variable initialization with a sentence and  a comment when initialize the frame counter and then refer to that here. 

I think diff syntax would probably be ok here but let me try the other approach first. As a new reader with fresh eyes on all this I think that would make more sense to me.

**Comment by ISSOtm** [comment]
> I can take care of this for you if I have permission to edit your fork.

Since you don't have upstream push access (something maybe we should consider? cc @avivace), an alternative is to suggest the modifications using review comments (they have an exclusive "suggest changes" feature); then it's "only" a two-click operation for the author.

File: src/part2/input.md [hunk 1]
```diff
@@ -29,6 +29,12 @@ First, let's set aside some room for the two variables that `UpdateKeys` will us
 Each variable must reside in RAM, and not ROM, because ROM is "Read-Only" (so you can't modify it).
 Additionally, each variable only needs to be one byte large, so we use `db` ("Define Byte") to reserve one byte of RAM for each.
 
+Before we read these variables we will also want to initialize them. We can do that below our initialization  of `wFrameCounter`.
```
Comment (ISSOtm) [inline src/part2/input.md:32]:
```suggestion
Before we read these variables we will also want to initialize them.
We can do that below our initialization  of `wFrameCounter`.
```

Just a style nitpick.

File: unbricked/bricks/main.asm [hunk 2]
```diff
@@ -90,8 +90,11 @@ ClearOam:
 	ld a, %11100100
 	ld [rOBP0], a
 
+	; Initialize variables
```
Comment (ISSOtm) [inline unbricked/bricks/main.asm:93]:
```suggestion
	; Initialize global variables
```
I think it's important to highlight what variables need to be initialised at this point. (A good rule of thumb is "initialise as close as possible to the point of usage".)

(Note: this change would have to be applied to the other instances as well.)

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Thank you! The change's contents look good to me; I would just like to tweak a few things before accepting it.

Though, @eievui5 mentioned that the chapter needs something rewritten? If you could look into that, then I wouldn't mind integrating the fix into this PR; but if you can't or don't want to in the next few weeks or so, then we can open a separate issue for it.

**Review by avivace — APPROVED** [review]

**Comment by avivace** [comment]
> Thank you! The change's contents look good to me; I would just like to tweak a few things before accepting it.
> 
> Though, @eievui5 mentioned that the chapter needs something rewritten? If you could look into that, then I wouldn't mind integrating the fix into this PR; but if you can't or don't want to in the next few weeks or so, then we can open a separate issue for it.

I'd suggest we merge this and limit the scope of this PR. Evie also already has "maintain" access to this repository.

**Review by ISSOtm — APPROVED** [review]
Then let's merge this improvement as it is :)

I'll let @eievui5 create an issue or PR with the changes she deems appropriate ^^

**Comment by ISSOtm** [comment]
Thank you @kav!

**Pull request merged**